src/UIHandler.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
+
+#include "cursesdef.h"
+#include "output.h"
+
+#include "enums.h"
+#include <memory>
+
+enum class Sizes {
+    Minimum,
+    Prefered
+};
+
+class UIUtils
+{
+    public:
+        static void DrawBorder( WINDOW *wf_win, point offset, point m_thisSize );
+        static void DrawTab( WINDOW *wf_win, point offset, int tabOffset, bool tabActive,
+                             std::string text );
+};
+
+class UIPanel
+{
+    public:
+        virtual std::vector<std::shared_ptr<UIPanel>> GetChild() const = 0;
+        virtual void AddChild( std::shared_ptr<UIPanel> panel ) = 0;
+        virtual void RemoveChild( size_t index ) = 0;
+
+        virtual point RequestedSize( Sizes sizes ) = 0;
+        virtual int SetSize( point size ) = 0;
+
+        virtual void DrawEverything( WINDOW *wf_window, point offset ) = 0;
+};
+
+class UIPaddingPanel : public UIPanel
+{
+    public:
+        UIPaddingPanel( bool drawBorder );
+        std::vector<std::shared_ptr<UIPanel>> GetChild() const;
```

Comment:
Btw: if you override a function, add the `override` keyword to it: `void GetChild() const override;`